modules/iam-role-for-service-accounts-eks/policies.tf

```
@@ -422,6 +422,50 @@ resource "aws_iam_role_policy_attachment" "efs_csi" {
   policy_arn = aws_iam_policy.efs_csi[0].arn
 }
 
+################################################################################
+# S3 CSI Driver Policy
+################################################################################
+
+#https://github.com/awslabs/mountpoint-s3/blob/main/doc/CONFIGURATION.md#iam-permissions
+data "aws_iam_policy_document" "mountpoint_s3_csi" {
+  count = var.create_role && var.attach_mountpoint_s3_csi_policy ? 1 : 0
+
+  statement {
+    sid       = "MountpointFullBucketAccess"
+    actions   = ["s3:ListBucket"]
+    resources = coalescelist(var.mountpoint_s3_csi_bucket_arns, ["arn:${local.partition}:s3:::*"])
+  }
+
+  statement {
+    sid = "MountpointFullObjectAccess"
+    actions = [
+      "s3:GetObject",
+      "s3:PutObject",
+      "s3:AbortMultipartUpload",
+      "s3:DeleteObject"
+    ]
+    resources = coalescelist(var.mountpoint_s3_csi_path_arns, ["arn:${local.partition}:s3:::*/*"])
```

This would give quite permissive access by default, so instead we will rely on users to provide the paths
```suggestion
    resources = var.mountpoint_s3_csi_path_arns
```